include/rocksdb/utilities/write_batch_with_index.h

```diff
@@ -176,6 +176,11 @@ class WriteBatchWithIndex : public WriteBatchBase {
   Iterator* NewIteratorWithBase(ColumnFamilyHandle* column_family,
                                 Iterator* base_iterator,
                                 const ReadOptions* opts = nullptr);
+  Iterator* NewIteratorWithBase(DB* db, Iterator* base_iterator,
```

Comment:
When is the DB parameter needed? (This is public API.)

Reply:
The DB option is used similarly to the DB option for "GetFromBatchAndBase".  Without the DB option, if you see a Merge without a preceding Put/Delete, you cannot get the "initial" value for the key without the DB.  If the other signatures are used (without the DB) and a Merge is encountered without a root, then a "MergeInProgress" error is returned.

I can update the comment appropriately when I figure out the wording...

Reply:
Although I don't know this area well, the rest of the PR LGTM, seems like a clear improvement even if it's not perfect.

Someone questioned whether it's a good idea for WBWI to effectively re-implement all the DB logic (without persistence, etc.). I believe it's potentially useful for finding bugs on both sides if we compare them.